lib/pdf_fill/filler.rb

```diff
@@ -138,10 +138,12 @@ def process_form(form_id, form_data, form_class, file_name_extension, fill_optio
       folder = 'tmp/pdfs'
       FileUtils.mkdir_p(folder)
       file_path = "#{folder}/#{form_id}_#{file_name_extension}.pdf"
-      start_page = form_class.const_defined?(:START_PAGE) ? form_class::START_PAGE : 1
-      hash_converter = HashConverter.new(form_id.sub(/V2\z/, ''), form_class.date_strftime,
-                                         fill_options.fetch(:extras_redesign, false), start_page)
-
+      extras_generator = ExtrasGenerator.new(
+        extras_redesign: fill_options.fetch(:extras_redesign, false),
+        form_name: form_id.sub(/V2\z/, ''),
+        start_page: form_class::START_PAGE, sections: form_class::SECTIONS
+      )
+      hash_converter = HashConverter.new(form_class.date_strftime, extras_generator)
```

Comment:
This redesign effort was adding a lot of new configuration arguments to `ExtrasGenerator.new`. I moved that call out of HashConverter into here, so that HashConverter doesn't need to pass along all those rendering-related arguments.